examples/move/big_vector/sources/big_vector.move

```diff
@@ -0,0 +1,1274 @@
+                // fix-up.
+                vals.destroy_empty();
+
+                self.root_id = NO_SLICE;
+            } else {
+                let mut root: Slice<u64> = df::remove(&mut self.id, root_id);
+                self.root_id = root.vals.pop_back();
+                self.depth = self.depth - 1;
+            }
+        };
+
+        val
+    }
+
+    /// Remove the elements between `lo` (inclusive) and `hi`
+    /// (exclusive) from `self`.
+    public fun remove_range<E: store + drop>(
+        _self: &mut BigVector<E>,
+        _lo: u128,
+        _hi: u128,
+    ) {
+        abort 0
+    }
+
+    /// Remove elements from `self` at the indices in `keys`,
+    /// returning the associated values.
+    ///
+    /// Aborts if any of the keys are not found.
+    public fun remove_batch<E: store>(
+        _self: &mut BigVector<E>,
+        _keys: vector<u128>,
+    ): vector<E> {
+        abort 0
+    }
+
+    // === SliceRef ===
+
+    /// Find the slice that contains the key-value pair for `key`,
+    /// assuming it exists in the data structure. Returns the
+    /// reference to the slice and the local offset within the slice
+    /// if it exists, aborts with `ENotFound` otherwise.
+    public fun slice_around<E: store>(
+        self: &BigVector<E>,
+        key: u128,
+    ): (SliceRef, u64) {
+        if (self.root_id == NO_SLICE) {
+            abort ENotFound
+        };
+
+        let (ix, leaf, off) = self.find_leaf(key);
+
+        if (off >= leaf.keys.length()) {
+            abort ENotFound
+        } else if (key != leaf.keys[off]) {
+            abort ENotFound
+        };
+
+        (SliceRef { ix }, off)
+    }
+
+    /// Find the slice that contains the key-value pair corresponding
+    /// to the next key in `self` at or after `key`. Returns the
+    /// reference to the slice and the local offset within the slice
+    /// if it exists, or (NO_SLICE, 0), if there is no matching
+    /// key-value pair.
+    public fun slice_following<E: store>(
+        self: &BigVector<E>,
+        key: u128,
+    ): (SliceRef, u64) {
+        if (self.root_id == NO_SLICE) {
+            return (SliceRef { ix: NO_SLICE }, 0)
+        };
+
+        let (ix, leaf, off) = self.find_leaf(key);
+        if (off >= leaf.keys.length()) {
+            (leaf.next(), 0)
+        } else {
+            (SliceRef { ix }, off)
+        }
+    }
+
+    /// Borrow a slice from this vector.
+    public fun borrow_slice<E: store>(
+        self: &BigVector<E>,
+        ref: SliceRef,
+    ): &Slice<E> {
+        df::borrow(&self.id, ref.ix)
+    }
+
+    /// Borrow a slice from this vector, mutably.
+    public fun borrow_slice_mut<E: store>(
+        self: &mut BigVector<E>,
+        ref: SliceRef,
+    ): &mut Slice<E> {
+        df::borrow_mut(&mut self.id, ref.ix)
+    }
+
+    // === Receiver function aliases ===
+
+    public use fun slice_is_null as SliceRef.is_null;
+    public use fun slice_is_leaf as Slice.is_leaf;
```

Comment:
This is a reminder that we need to update GitHub's highlighting for Move, right?